### DISCUSSION TITLE
Injecting seating data with a custom plugin

### DISCUSSION THREAD

**Discussion by moritzlerch** [pretix/pretix#3306] [Q&A]
Hey, 

were currently developing an algorithm that automatically assigns the seats for group bookings and desired seat neighbours in a big venue. I saw the commercially available seating plugin offering a setting that doesn't allow the users to select the seats on their on but to actually only make these settings available from the backend. 

Would it be possible to write a custom plugin that injects the seating data (desirably without the actual need to import a seating plan) per order so that the customers can see their assigned seats in the order page and on their pdf ticket?

All the best and many thanks in advance :)

**Comment by raphaelm** [comment]
Yes, you could do that, all you need to do to make the seatings "chosen not by the user" would be setting ``event.settings.seating_choice = false`` when your plugin is enabled

**Comment by moritzlerch** [comment]
But how would I implement the assignment of seats? There is no documentation about the seats. Furthermore, is there any way of getting around of having to import the seating plan (as a json)? Can I natively assign seats without having them checked against the seating plan?

**Reply by raphaelm** [reply]
You can assign seats using 

```
ocm = OrderChangeManager(order)
ocm.change_seat(orderposition, seat)
ocm.commit()
```

(or similar, typed on my phone)

But yeah you need to have `Seat` objects first which usually are created through the import of a seating plan.

It's certainly not easy to build something with seating without the commercial seating plugin since everything but the most basic data model and logic is missing. But it can be done of course.
